sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetPartitionReaderFactory.scala

```diff
@@ -92,8 +93,13 @@ case class ParquetPartitionReaderFactory(
     if (aggregation.isEmpty) {
       ParquetFooterReader.readFooter(conf, filePath, SKIP_ROW_GROUPS)
     } else {
+      val split = new FileSplit(file.toPath, file.start, file.length, Array.empty[String])
       // For aggregate push down, we will get max/min/count from footer statistics.
-      ParquetFooterReader.readFooter(conf, filePath, NO_FILTER)
```

Comment:
@huaxingao Hi~when reviewing the code, I have a point of doubt, why not take the RowGroup metrics information in the current file range here?
Here is a example in `SpecificParquetRecordReaderBase`
https://github.com/apache/spark/blob/61035129a354d0b31c66908106238b12b1f2f7b0/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java#L96-L102

Reply:
@yabola Thanks for the ping. I was assuming that we always need to read the whole `PartitionedFile`, i.e. the start is always 0 and the length is always the file length. Maybe it's safer to add the range.

Reply:
@sunchao What do you think?

Reply:
@LuciferYang @huaxingao  oh~ thank you! Yes, using `NO_FILTER` is no problem here.
https://github.com/apache/spark/blob/df2e2516188b46537349aa7a5f279de6141c6450/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScan.scala#L50-L57

 But do you think it will be always safer to use file range? and I have made some changes here, I want to unify them  https://github.com/apache/spark/pull/39950